packages/components/src/components/common/inputtag/inputBadge.tsx

```diff
@@ -0,0 +1,79 @@
+import { forwardRef, type ElementRef } from 'react';
+import { useField } from '../field/field';
+import { IvyIcon } from '../icon/icon';
+import { inputBadge, inputBadgeIcon, inputBadgeLine, inputBadgeOutput } from './inputBadge.css';
+
+import type { IvyIcons } from '@axonivy/ui-icons';
+
+export type InputBadgeProps = {
+  value: string;
+  badgeRegex: BadgeRegex[];
+};
+
+type BadgeRegex = { delimiter: { start: string; end: string }; icon: IvyIcons; badgeTextGen: (text: string) => string };
+
+export type BadgeTextProps = { icon: IvyIcons | undefined; text: string };
+
+const Badge = ({ badge }: { badge: BadgeTextProps }) => {
```

Comment:
why wrap it again in a badge object and not directly use the properties?
```suggestion
const Badge = ({ text, icon }: BadgeTextProps ) => {
```